### PR TITLE
Camp Finder: Backport from YGBW (extend for Activity Finder)

### PR DESCRIPTION
**Pull request by jediandrey**
Original Issue, this PR is going to fix: REPLACE WITH A LINK TO ISSUE ( publicly available )

Make sure these boxes are checked before asking for review of your pull request - thank you!

If there is a new feature or this is a bug fix - use 8.x-2.x branch. We'll tag for release if the bug is critical asap or tag for release next bug fix release until critical issue arrived.

## Steps for review

- [ ] create 2 new pages, once for camp finder, one for results as usually for activity finder
- [ ] On Camp Finder check "Show filters opened by default"
- [ ] On Results check "Show weeks" filter
- [ ] ensure you indexed content with a new field "Weeks"

## General checks
- [ ] All coding styles are fulfilled and there are no any issues reported by CodeSniffer. See [Code of Conduct](https://github.com/ymcatwincities/openy/wiki/Open-Y-Code-of-Conduct-and-Best-Practices).
- [ ] [Documentation](https://github.com/ymcatwincities/openy/tree/8.x-1.x/docs) has been updated according to PR changes.
- [ ] [Steps for review](https://github.com/ymcatwincities/openy/pull/94#issue-204580200) have been provided according to PR changes. <br/>Steps for review
- [ ] Make sure you've provided all necessary hook\_update\_N to [support upgrade path](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Upgrade%20path.md).
- [ ] Make sure your git email is associated with account on drupal.org, otherwise you won't get commits there. <br/>drupal.org email
- [ ] If you would like to get credits on drupal.org, [check documentation](https://github.com/ymcatwincities/openy/blob/8.x-1.x/docs/Development/Contributing.md#drupalorg-credits).

Thank you for your contribution!

### CODE REVIEW

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2133/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2133/

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2133/

File: modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install [hunk 1]
```diff
@@ -17,3 +17,19 @@ function openy_paragraph_activity_finder_search_update_8002() {
     'core.entity_form_display.paragraph.activity_finder_search.default'
   ]);
 }
+
+/**
+ * Update configs.
+ */
+function openy_paragraph_activity_finder_search_update_8003() {
+  // We exported this configs to sync, so no need to import this in update.
+  return;
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install]: Why are we adding this update function to Open Y?

File: modules/custom/openy_activity_finder/modules/openy_prgf_camp_finder/openy_prgf_camp_finder.install [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?php
+
+/**
+ * @file
+ * Contains install and update routines.
+ */
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/modules/openy_prgf_camp_finder/openy_prgf_camp_finder.install:6]: Apparently this file is not needed.

File: modules/custom/openy_activity_finder/openy_activity_finder.libraries.yml [hunk 3]
```diff
@@ -18,6 +18,24 @@ openy_activity_finder:
     - openy_system/moment
     - core/jquery.cookie
 
+openy_camp_finder:
+  version: 0.1
+  js:
+    openy_cf_vue_app/dist/main.js: {}
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/openy_activity_finder.libraries.yml]: It looks like this file is minified - please add appropriate flags.

File: modules/custom/openy_activity_finder/openy_cf_vue_app/dist/main.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
...)}var ia={model:Qi,show:{bind:function(t,e,n){var r=e.value,i=(n=ra(n)).data&&n.data.transition,a=t.__vOriginalDisplay="none"===t.style.display?"":t.style.display;r&&i?(n.data.show=!0,Hi(n,function(){t.style.display=a})):t.style.display=r?a:"none"},update:function(t,e,n){var r=e.value;!r!=!e.oldValue&&((n=ra(n)).data&&n.data.transition?(n.data.show=!0,r?Hi(n,function(){t.style.display=t.__vOriginalDisplay}):Vi(n,function(){t.style.display="none"})):t.style.display=r?t.__vOriginalDisplay:"none")},unbind:function(t,e,n,r,i){i||(t.style.display=t.__vOriginalDisplay)}}},aa={name:String,appear:Boolean,css:Boolean,mode:String,type:String,enterClass:String,leaveClass:String,enterToClass:String,leaveToClass:String,enterActiveClass:String,leaveActiveClass:String,appearClass:String,appearActiveClass:String,appearToClass:String,duration:[Number,String,Object]};function oa(t){var e=t&&t.componentOptions;return e&&e.Ctor.options.abstract?oa(Ke(e.children)):t}function sa(t){var e={},n=t.$options;for(var r in n.propsData)e[r]=t[r];var i=n._parentListeners;for(var a in i)e[C(a)]=i[a];return e}function ca(t,e){if(/\d-keep-alive$/.test(e.tag))return t("keep-alive",{props:e.componentOptions.propsData
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/openy_cf_vue_app/dist/main.js]: Vue and Vue-router packaged here - please exclude them from this artifact.

File: modules/custom/openy_activity_finder/src/Plugin/search_api/processor/PartsOfDay.php [hunk 5]
```diff
@@ -1,89 +0,0 @@
-<?php
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/src/Plugin/search_api/processor/PartsOfDay.php]: Why did you remove this file?

File: modules/custom/openy_activity_finder/templates/openy-camp-finder-program-search.html.twig [hunk 6]
```diff
@@ -0,0 +1,24 @@
+{#
+/**
+ * @file
+ *
+ * Available variables:
+ * - data: contains all the data.
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/templates/openy-camp-finder-program-search.html.twig]: This comment needs some love; at least the list of variables should be updated

File: modules/custom/openy_activity_finder/openy_cf_vue_app/components/CampFinder.vue [hunk 7]
```diff
@@ -0,0 +1,514 @@
...
+        }
+
+        if (typeof this.$route.query.initial != 'undefined') {
+          let initialStepGet = decodeURIComponent(this.$route.query.initial);
+          if (initialStepGet !== '') {
+            this.initialStep = initialStepGet;
+          }
+        }
+      },
+      getHumanReadableAppliedFilters() {
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/openy_cf_vue_app/components/CampFinder.vue:362]: Why isn't this method in "computed" section? It seems like it depends only on component state and should be there in computed properties.

File: modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock [hunk 8]
```diff
@@ -0,0 +1,5025 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (duozersk) [inline modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock:1]: You are instructing to use npm in the README; but here we have yarn.lock - it doesn't look good. Please use npm, commit the package-lock.json and remove this yarn.lock file.

File: modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install (hunk 1 above)
Comment (danylevskyi) [reply modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install]: I believe some YMCA already using AF.
This is hook update.
@duozersk ?

**Review by danylevskyi — COMMENTED** [review]
Had a quick review.
Nothing bad found.

File: modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install (hunk 1 above)
Comment (jediandrey) [reply modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install]: Yes, many YMCA are using AF, lets keep it (return is ygbw specific, I'm going to remove it)

File: modules/custom/openy_activity_finder/src/Plugin/search_api/processor/PartsOfDay.php (hunk 5 above)
Comment (jediandrey) [reply modules/custom/openy_activity_finder/src/Plugin/search_api/processor/PartsOfDay.php]: Accidentally during massive merge, returned

File: modules/custom/openy_activity_finder/openy_cf_vue_app/dist/main.js (hunk 4 above)
Comment (jediandrey) [reply modules/custom/openy_activity_finder/openy_cf_vue_app/dist/main.js]: done

File: modules/custom/openy_activity_finder/openy_cf_vue_app/components/CampFinder.vue (hunk 7 above)
Comment (jediandrey) [reply modules/custom/openy_activity_finder/openy_cf_vue_app/components/CampFinder.vue:362]: I prefer to keep it in methods section for now

File: modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock (hunk 8 above)
Comment (jediandrey) [reply modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock:1]: updated README

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2137/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2137/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2137/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2137/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2137/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2137/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2137/blogWCAG2AAhtmlcs.txt

**Comment by bjwilkins** [comment]
Build comment file: 
--------------

Check Open Y Installation Wizard at http://install.openy.cibox.tools/build2138/install.php

--------------
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://openy.cibox.tools:8080/job/PR_BUILDER_COMPOSER/2138/

**Comment by bjwilkins** [comment]
Acessibility Sniffer: front page checking WCAG2AA http://openy.cibox.tools/build2138/frontWCAG2AAhtmlcs.txt
Acessibility Sniffer: join page checking WCAG2AA http://openy.cibox.tools/build2138/joinWCAG2AAhtmlcs.txt
Acessibility Sniffer: locations page checking WCAG2AA http://openy.cibox.tools/build2138/locationsWCAG2AAhtmlcs.txt
Acessibility Sniffer: schedules page checking WCAG2AA http://openy.cibox.tools/build2138/schedulesWCAG2AAhtmlcs.txt
Acessibility Sniffer: blog page checking WCAG2AA http://openy.cibox.tools/build2138/blogWCAG2AAhtmlcs.txt

File: modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install (hunk 1 above)
Comment (duozersk) [reply modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install]: It has a "return;" statement as the first executable line... so apparently it does nothing.

Comment (jediandrey) [reply modules/custom/openy_activity_finder/modules/openy_paragraph_activity_finder_search/openy_paragraph_activity_finder_search.install]: It has not anymore

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2137/

**Comment by bjwilkins** [comment]
Open Y Complete profile installed at http://openy.cibox.tools/build2138/

File: modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock (hunk 8 above)
Comment (duozersk) [reply modules/custom/openy_activity_finder/openy_cf_vue_app/yarn.lock:1]: I would love to see some standardization here... npm vs yarn - npm is more common and we should have one way to simplify maintenance and CI environments. I would suggest to standardize on npm.
And we can't use both at the same time - there should be no application with both package-lock.json and yarn.lock files (like we have in AF3 now).
//cc @podarok
